### DISCUSSION TITLE
Using Open Sea Map

### DISCUSSION THREAD

**Discussion by andy130342** [fleaflet/flutter_map#1174] [Q&A] [answered]
Has anyone already managed to use Open **Sea** Map in flutter_map? If not, can anyone explain the necessary steps to get there?

**Comment by ibrierley** [comment]
I think it should work ok, just add it as a layer on top of your normal tile layer iirc. It was a long time ago since I saw it, so can't remember exactly.

**Reply by andy130342** [reply]
Maybe this is a stupid question, but I think it should look somehow like this:
layers: [
          TileLayerOptions(
            urlTemplate: "https://{s}.tiles.openseamap.org/seamark/{z}/{x}/{y}.png",
            subdomains: ['a', 'b', 'c'],
          ),
But this seems not to be the right URL. Do you know the right URL or where I could get it?

**Comment by ibrierley** [comment, marked as answer]
I don't think it supports subdomains, try...
```
layers: [
TileLayerOptions(
urlTemplate: "https://tiles.openseamap.org/seamark/{z}/{x}/{y}.png",
),
```

**Reply by andy130342** [reply]
Thanks a lot, that works.

**Comment by tobiasht** [comment]
I am unable to see the base map layer, only the overlays. What am I missing?

**Reply by ibrierley** [reply]
You're missing posting some code, so people may have an idea what you're missing :).

**Reply by tobiasht** [reply]
Sorry, here is my code.

```dart
class OpenSeaMap extends StatelessWidget {
  const OpenSeaMap({super.key});

  @override
  Widget build(BuildContext context) {
    return FlutterMap(
      options: MapOptions(),
      children: [
        TileLayer(
          urlTemplate: "https://tiles.openseamap.org/seamark/{z}/{x}/{y}.png",
        ),
      ],
    );
  }
}
```

And this is what i see. 

![IMG_4754](https://github.com/fleaflet/flutter_map/assets/20655261/b0fb4a06-88c3-46d4-863a-3ad9f04d6e2c)

**Reply by ibrierley** [reply]
I don't think you're actually including any other layer in your code ? Include another tile layer first with which ever map base you want ?

**Reply by tobiasht** [reply]
I´ve already tried to have another map as the base, but the gray background covers it unfortunately. 
But maybe I am doing it wrong?  

```dart
class OpenSeaMap extends StatelessWidget {
  const OpenSeaMap({super.key});

  @override
  Widget build(BuildContext context) {
    return FlutterMap(
      options: MapOptions(),
      children: [
        TileLayer(
          urlTemplate: 'https://tile.openstreetmap.org/{z}/{x}/{y}.png',
        ),
        TileLayer(
          urlTemplate: "https://tiles.openseamap.org/seamark/{z}/{x}/{y}.png",
        ),
      ],
    );
  }
}
```

**Reply by JaffaKetchup** [reply]
Try setting the background color of the sea map to be transparent. (https://docs.fleaflet.dev/layers/tile-layer#:~:text=It%20is%20possible%20to%20use%20more%20than%20one%20tile%20layer%2C%20and%20can%20be%20used%20with%20transparency/opacity.)

**Reply by tobiasht** [reply]
Thank you so much for the help. That's a parameter i wasn't aware of. Kinda weird it's not set to transparent by default, but it's probably a reason for it. In the end i just had to add the following line:

```dart
backgroundColor: Colors.transparent,
```

Final code below
```dart
class OpenSeaMap extends StatelessWidget {
  const OpenSeaMap({super.key});

  @override
  Widget build(BuildContext context) {
    return FlutterMap(
      options: MapOptions(),
      children: [
        TileLayer(
          urlTemplate: 'https://tile.openstreetmap.org/{z}/{x}/{y}.png',
        ),
        TileLayer(
          backgroundColor: Colors.transparent,
          urlTemplate: "https://tiles.openseamap.org/seamark/{z}/{x}/{y}.png",
        ),
      ],
    );
  }
}
```

**Reply by JaffaKetchup** [reply]
Great!

It's grey as default because:
* That's 'traditional' for mapping libraries instead of white, and resolves confusion about whether the map widget is working at all
* Most users will have one tile layer, so the above applies

Maybe we could do some fancy logic to set background color to transparent if it's not the first layer in the `children` list. Shouldn't be too hard?
